segway/input_master.py

```diff
@@ -209,131 +181,303 @@ def get_template_component_suffix(self, component_number):
         else:
             return "_component{}".format(component_number)
 
-    def generate_tmpl_mappings(self):
-        # need segnames because in the tied covariance case, the
-        # segnames are replaced by "any" (see .make_covar_spec()),
-        # and only one mapping is produced
-        num_subsegs = self.num_subsegs
-
-        track_groups = self.track_groups
-        num_track_groups = self.num_track_groups
-
-        for seg_index, segname in enumerate(self.make_segnames()):
-            seg_offset = num_track_groups * num_subsegs * seg_index
-
-            for subseg_index, subsegname in enumerate(self.make_subsegnames()):
-                subseg_offset = seg_offset + (num_track_groups * subseg_index)
-
-                for track_group_index, track_group in enumerate(track_groups):
-                    track_offset = subseg_offset + track_group_index
-                    head_trackname = track_group[0].name
-
-                    # XXX: change name of index to track_offset in templates
-                    # XXX: change name of track_index to track_group_index
-                    yield dict(seg=segname, subseg=subsegname,
-                               track=head_trackname, seg_index=seg_index,
-                               subseg_index=subseg_index,
-                               track_index=track_group_index,
-                               index=track_offset,
-                               distribution=self.distribution)
-
-    def make_data(self):
+    def get_head_track_names(self):
         """
-        override this in subclasses
-
-        returns: container indexed by (seg_index, subseg_index, track_index)
+        Returns list containing the first track name in each track group.
         """
-        return None
-
-    def generate_objects(self):
+        head_track_names = []
+        for group in self.track_groups:
+            head_track_names.append(group[0].name)
+        return head_track_names
+      
+    def generate_collection_names(self):
         """
-        returns: iterable of strs containing GMTK parameter objects starting
-        with names
+        Generate names of NameCollection objects. 
         """
-        substitute = Template(self.object_tmpl).substitute
-
-        data = self.make_data()
-
-        for mapping in self.generate_tmpl_mappings():
-            track_index = mapping["track_index"]
+        head_track_names = self.get_head_track_names()
+        names = []
+        for name in head_track_names:
+            names.append("collection_seg_{}".format(name))
```

Comment:
For all the GMTK object name format strings, you should define them as constants above. e.g. `NAME_COLLECTION_FORMAT_STRING = "collection_seg_{}"`